notebooks/tutorial/7 - Getting data from the ADX.ipynb

```diff
@@ -0,0 +1,705 @@
+    "                    \"Statement\": [\n",
+    "                        {\n",
+    "                            \"Effect\": \"Allow\",\n",
+    "                            \"Principal\": {\"Service\": \"dataexchange.amazonaws.com\"},\n",
+    "                            \"Action\": [\"s3:PutObject\", \"s3:PutObjectAcl\"],\n",
+    "                            \"Resource\": f\"arn:aws:s3:::{bucket}/*\",\n",
+    "                            \"Condition\": {\"StringEquals\": {\"aws:SourceAccount\": account_id}},\n",
+    "                        }\n",
+    "                    ],\n",
+    "                }\n",
+    "            ),\n",
+    "        )\n",
+    "        \n",
+    "        if days_to_keep_data_on_s3 is None:\n",
+    "            client.put_bucket_lifecycle_configuration(\n",
+    "                Bucket=bucket,\n",
+    "                LifecycleConfiguration={\n",
+    "                    \"Rules\": [\n",
+    "                        {\n",
+    "                            \"Expiration\": {\n",
+    "                                \"Days\": days_to_keep_data_on_s3,\n",
+    "                            },\n",
+    "                            \"Status\": \"Enabled\",\n",
+    "                            \"Filter\": {\"Prefix\": ''}\n",
+    "                        },\n",
+    "                    ],\n",
+    "                },\n",
+    "            )\n",
+    "        \n",
+    "        client.put_public_access_block(\n",
+    "            Bucket=bucket,\n",
+    "            PublicAccessBlockConfiguration={\n",
+    "                \"BlockPublicAcls\": True,\n",
+    "                \"IgnorePublicAcls\": True,\n",
+    "                \"BlockPublicPolicy\": True,\n",
+    "                \"RestrictPublicBuckets\": True,\n",
+    "            },\n",
+    "        )\n",
+    "    else:\n",
+    "        print(f\"bucket {bucket} already exists.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "50124c08",
+   "metadata": {},
+   "source": [
+    "## Transfer latest revision from ADX to S3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a3fe4e66",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "if transfer_option == \"latest\" and agreed_to_terms:\n",
+    "    export_single_adx_dataset_revision_to_s3(\n",
+    "        bucket, dataset_id, prefix=prefix\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a562d6a5",
+   "metadata": {},
+   "source": [
+    "## Transfer date range from ADX to S3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9892e84a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if transfer_option == \"range\" and agreed_to_terms:\n",
+    "    export_multiple_adx_dataset_revisions_to_s3(\n",
+    "        bucket,\n",
+    "        dataset_id,\n",
+    "        prefix=prefix,\n",
+    "        start_date=single_date,\n",
```

Comment:
Where is `single_date` set? Is this a bug?

Reply:
yes

Reply:
Fix'd